### ISSUE TITLE
Linkage warnings about same member name on Xcode for same-name .cpp files

### ISSUE THREAD

**Issue by akien-mga** [godotengine/godot#24419]
**Godot version:**
Master (4b6e7b5)

**OS/device including version:**
macOS, Xcode 9.4.1 (Travis)

**Issue description:**
The following warnings are raised by Xcode's `libtool` when linking an iOS `arm64` build:
```
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool -static -o "/Users/travis/build/akien-mga/godot/bin/libgodot.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/platform/iphone/libiphone.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/main/libmain.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/main/tests/libtests.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/platform/libplatform.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/drivers/libdrivers.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/scene/libscene.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/servers/libservers.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a" "/Users/travis/build/akien-mga/godot/modules/freetype/libfreetype_builtin.iphone.debug.arm64.a"
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (aabb.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(aabb.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(aabb.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (array.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(array.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(array.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (base64.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(base64.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(base64.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (color.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(color.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(color.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (compress.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(compress.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(compress.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (cpu.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(cpu.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(cpu.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (debug.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(debug.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(debug.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (dictionary.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(dictionary.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(dictionary.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (gdnative.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(gdnative.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(gdnative.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (global_defaults.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(global_defaults.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(global_defaults.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (info.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(info.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(info.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (info.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(info.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(info.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (internal.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(internal.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(internal.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (main.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/main/libmain.iphone.debug.arm64.a(main.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/platform/iphone/libiphone.iphone.debug.arm64.a(main.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (md5.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(md5.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(md5.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (mdct.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(mdct.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(mdct.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (node_path.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(node_path.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(node_path.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (plane.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(plane.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(plane.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (quat.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(quat.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(quat.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (rect2.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(rect2.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(rect2.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (register_types.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(register_types.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (rid.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(rid.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(rid.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (sha256.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(sha256.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(sha256.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (threading.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(threading.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(threading.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (transform.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(transform.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(transform.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (variant.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(variant.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(variant.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (vector2.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(vector2.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(vector2.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
/Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/libtool: warning same member name (vector3.iphone.debug.arm64.o) in output file used for input files: /Users/travis/build/akien-mga/godot/modules/libmodules.iphone.debug.arm64.a(vector3.iphone.debug.arm64.o) and: /Users/travis/build/akien-mga/godot/core/libcore.iphone.debug.arm64.a(vector3.iphone.debug.arm64.o) due to use of basename, truncation and blank padding
```
(Travis log: https://travis-ci.org/akien-mga/godot/jobs/468953696)

Is there a setting we can use so that `libtool` uses full paths instead of basenames to differentiate files with the same basename?

**Comment by follower** [comment]
----

**TL;DR:** Consider using `ar` instead. (via https://github.com/grpc/grpc-swift/pull/447#issuecomment-487127319)

----

It seems this warning is in honor of the rich history of Unix... :D

From <https://opensource.apple.com/source/cctools/cctools-949.0.1/misc/libtool.c.auto.html>:

```c
/*
 * warn_duplicate_member_names() generates a warning if two members end up with
 * the same ar_name.  This is only a warning because ld(1) and this program
 * has no problems with it.  Only if ar(1) were used to extract the files
 * would this be a problem (even the 4.4bsd ar(1) using long names can
 * get hosed by base names,  the 4.3bsd ar(1) can't handle full 16 character
 * ar_names).
 */
static
void
warn_duplicate_member_names(
void)
{
```

And as mentioned in https://github.com/grpc/grpc-swift/pull/447#issuecomment-487127319 the same code (& executable?) is used for `ranlib` & `libtool` but only `libtool` generates the warning:

```c
/* if this is libtool warn about duplicate member names */
	if(cmd_flags.ranlib == FALSE)
	    warn_duplicate_member_names();
```

So, it seems the two options to silence the (seemingly unnecessary) warnings are:

 * Use `ar` (together with `ranlib`?); or,

 * Rename source files to avoid the clash.

_[WIP: I've got some supporting links to add but in case I don't get back to this issue for a while the above is the general gist. :) ]_

**Comment by dmoody256** [comment]
seems the OBJSUFFIX or SHOBJSUFFIX could be easily used in a few specific cases to get different named obj file, for example:

```
core_env['OBJSUFFIX'] = '.core' + env['OBJSUFFIX']
core/aabb.cpp -> aabb.core.iphone.debug.arm64.o

gdnative_env['OBJSUFFIX'] = '.gdnative' + env['OBJSUFFIX']
core/aabb.cpp -> aabb.gdnative.iphone.debug.arm64.o
```

**Comment by akien-mga** [comment]
This was fixed by the replacement of GDNatice with GDExtension.
